users/views.py

```diff
@@ -41,4 +45,26 @@ def post(self, request):
             return JsonResponse({"message": "SUCCESS"}, status=201)
 
         except KeyError:
-            return JsonResponse({"message": "Key_Error"}, status=400)
\ No newline at end of file
+            return JsonResponse({"message": "Key_Error"}, status=400)
+
+class SigninView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            username = data['id']
+            password = data['password']
+            user = User.objects.get(username=username)
+
+            if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
+                access_token = jwt.encode({"user_id": user.id}, settings.SECRET_KEY, algorithm=settings.ALGORITHM)
+                
+                return JsonResponse({"access_token": access_token}, status=200)
+
+            else:
+                return JsonResponse({"message": "INVALID_USER"}, status=401)
```

Comment:
아래와 같이 if 문의 조건을 반대로 (잘못된 케이스) 로 바꿔준다면 지금처럼 if ... else ... 를 사용하여 로직의 흐름이 두갈래로 갈리는것보다 더 깔끔한 플로우를 만들어줄 수 있습니다 :)
```suggestion
            if not bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
                return JsonResponse({"message": "INVALID_USER"}, status=401)
            
            access_token = jwt.encode({"user_id": user.id}, settings.SECRET_KEY, algorithm=settings.ALGORITHM)
                
            return JsonResponse({"access_token": access_token}, status=200)
```

Reply:
리뷰주신 내용 확인 후 수정 완료하였습니다 😊